src/chain.h

```diff
@@ -192,7 +192,9 @@ class CBlockIndex
     uint32_t nNonce{0};
 
     //! (memory only) Sequential id assigned to distinguish order in which blocks are received.
-    int32_t nSequenceId{0};
+    //! Initialized to 1 when loading blocks from disk, except for blocks belonging to the best chain
+    //! which overwrite it to 0.
+    int32_t nSequenceId{1};
```

Comment:
In 6756af191f0:
nit: It might be more maintainable to define two constants, such as `ID_DISK-ONLY_BEST_CHAIN` and `ID_DISK-ONLY_FORK` (or something similar).